packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -926,7 +927,13 @@ function mountEffectImpl(fiberEffectTag, hookEffectTag, create, deps): void {
   hook.memoizedState = pushEffect(hookEffectTag, create, undefined, nextDeps);
 }
 
-function updateEffectImpl(fiberEffectTag, hookEffectTag, create, deps): void {
+function updateEffectImpl(
+  fiberEffectTag,
+  hookEffectTag,
+  noWorkUnmountFiberHookEffectTag,
```

Comment:
This name isn't the greatest either. I welcome better suggestions.

The purpose of the changes to this function is to retain the ability to distinguish between layout and passive effects even if a render has no dependency changes. Previously, we would not be able to distinguish between the two during unmount.